documentation/modules/about-hooks-for-migration-plans.adoc

```diff
@@ -0,0 +1,77 @@
+
+* Hooks that perform operations on VMs after migration are called _post-migration_ hooks. A post-migration hook is performed on a VM after it has been migrated to {virt}.
+
+[id="default-hook-image_{context}"]
+== Default hook image
+The default hook image for an {project-short} hook is `quay.io/konveyor/hook-runner`. The image is based on the Ansible Runner image with the addition of `python-openshift` to provide Ansible Kubernetes resources and a recent `oc` binary.
+
+[id="hook-execution_{context}"]
+== Hook execution
+An Ansible playbook that is provided as part of a migration hook is mounted into the hook container as a `configmap`. The hook container is run as a job on the desired cluster, using the default `ServiceAccount` in the `konveyor-forklift` namespace.
+
+
+When you add a hook, you must specify the namespace where the `Hook` CR is located, the name of the hook, and specify whether the hook is a pre-migration hook or a post-migration hook.
+
+[IMPORTANT]
+====
+In order for a hook to run on a VM, the VM must be started and available via SSH.
+====
+
+The illustration that follows shows the general process of using a migration hook. Specific procedures are described in xref:adding-migration-hook-via-ui_{context}[Adding a migration hook to a migration plan using the {ocp} web console] and xref:adding-migration-hook-via-cli_{context}[Adding a migration hook to a migration plan using the CLI].
+
+// THE FIGURE WILL BE REPLACED
+
+.Adding a hook to a migration plan
+image::updated_hooks_figure_from_martin_cropped.png[Adding a hook to migration plan]
+
+*Process:*
+
+. Input your Ansible hook and credentials.
+
+.. Input an Ansible hook image to the {project-short} controller using either the UI or the CLI.
++
+* In the UI, specify the `ansible-runner` and enter the `playbook.yml` that contains the hook.
+* In the CLI, input the hook image, which specifies the playbook that runs the hook.
+
+.. If you need additional data to run the playbook inside the pod, such as SSH data, create a Secret that contains credentials for the VM. The Secret is not mounted to the pod, but is called by the playbook.
++
+[NOTE]
+====
+This Secret is not the same as the `Secret` CR that contains the credentials of your source provider.
+====
+
+. The {project-short} controller creates the `configmap`, which contains:
+
+** `workload.yml`, which contains information about the VMs.
+** `playbook.yml`, the raw string playbook you want to execute.
+** `plan.yml`, which is the `Plan` CR.
++
+The `configmap` contains the name of the VM and instructs the playbook what to do.
+
+. The {project-short} controller creates a job that starts the user specified image.
+.. Mounts the `configmap` to the container.
++
+The Ansible hook imports the Secret that the user previously entered.
+. The job runs a prehook or a posthook as follows:
```

Comment:
```suggestion
. The job runs a pre-migration hook or a post-migration hook as follows:
```
I would probably be explicit... as I do not like `prehook` and `posthook` (but this is your choice not mine)